### PR TITLE
Update VM controller error handling

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

Sometimes some resources might not exist yet and we see a huge amount of logs stating that an error created the VM metrics.

This PR:

- adds 5 seconds backoff between failed reconcile loops of VM controller to reduce the number of logs 
- removes 'error' wording from logs when the VM controller does not create the VM-related metrics, since it might not represent more serious problems, and might deceive users
- skips metric creation when VM-related resources are still in Provisioning status

**Which issue(s) this PR fixes**: 

jira-ticket: https://issues.redhat.com/browse/CNV-35806

**Special notes for your reviewer**:

**Release note**:

```release-note
Update VM controller error handling
```

### CODE REVIEW

**Comment by akrejcir** [comment]
If a PVC or PV does not exist, we want to try checking again? Will these resources be created eventually?

**Comment by machadovilaca** [comment]
> If the PVC or PV does not exist, do we want to try checking again? Will these resources be created eventually?

The PVC is associated to the VM, is just not created yet, the backoff ensure we will try the reconcile loop again in a few seconds

**Comment by akrejcir** [comment]
Instead of retrying later, can you check the `.status.printableStatus` field?
The status `Provisioning` is set when not all resources are created yet: 
https://github.com/kubevirt/kubevirt/blob/9d9885c655d465657f515c8f73c23b00c9f16898/staging/src/kubevirt.io/api/core/v1/types.go#L1511-L1513

If a VM has this `printableStatus`, it can be skipped. Later, if the `.status` filed changes, the VM will trigger reconciliation again.
This way we will avoid infinite looping while waiting for the resources to be created.

File: controllers/vm_controller.go [hunk 1]
```diff
@@ -65,13 +66,20 @@ func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
...
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: 5 * time.Second,
+		}, err
 	}
 
 	if err := r.setVmVolumesMetrics(ctx, &vm); err != nil {
-		r.log.Error(err, "Error setting vm volumes metrics", "vm", req.NamespacedName)
-		return ctrl.Result{}, err
+		r.log.Error(err, "could not set vm volumes metrics", "vm", req.NamespacedName)
```
Comment (sradco) [inline controllers/vm_controller.go:78]: Nit: `could` -> `Could`

File: controllers/vm_controller.go [hunk 2]
```diff
@@ -65,13 +66,20 @@ func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 
 			return ctrl.Result{}, nil
 		}
-		r.log.Error(err, "Error getting VM", "vm", req.NamespacedName)
-		return ctrl.Result{}, err
+
+		r.log.Error(err, "could not find VM", "vm", req.NamespacedName)
```
Comment (sradco) [inline controllers/vm_controller.go:70]: Nit: `could` -> `Could`

File: controllers/vm_controller.go [hunk 3]
```diff
@@ -65,13 +66,20 @@ func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
...
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: 5 * time.Second,
```
Comment (sradco) [inline controllers/vm_controller.go:73]: @machadovilaca  Why do we wait here 5sec and for the Volumes we wait 30sec?

Comment (machadovilaca) [reply controllers/vm_controller.go:73]: waiting time for the PVC to be create, but not needed anymore

**Comment by machadovilaca** [comment]
update @akrejcir

File: controllers/vm_controller.go [hunk 4]
```diff
@@ -65,13 +66,26 @@ func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
...
-		return ctrl.Result{}, err
+
+		r.log.Error(err, "Could not find VM", "vm", req.NamespacedName)
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: 5 * time.Second,
+		}, err
+	}
+
+	if vm.Status.PrintableStatus != kubevirtv1.VirtualMachineStatusProvisioning {
```
Comment (akrejcir) [inline controllers/vm_controller.go:77]: Typo: Should be `==`.

**Review by akrejcir — COMMENTED** [review]
Thanks. Can you add more text to the commit message?

**Comment by machadovilaca** [comment]
> Thanks. Can you add more text to the commit message?

done!

**Pull request merged**